flatcar-terraform-hetzner/server-configs/server1.yaml

```diff
@@ -0,0 +1,25 @@
+variant: flatcar
+version: 1.0.0
+
+# This is a simple NGINX example.
+# Replace the below with your own config.
+# Refer to https://www.flatcar.org/docs/latest/provisioning/config-transpiler/configuration/ for more information.
+
+systemd:
+  units:
+    - name: nginx.service
+      enabled: true
+      contents: |
+        [Unit]
+        Description=NGINX example
+        After=docker.service
+        Requires=docker.service
+        [Service]
+        TimeoutStartSec=0
+        ExecStartPre=-/usr/bin/docker rm --force nginx1
+        ExecStart=/usr/bin/docker run --name nginx1 --pull always --net host docker.io/nginx:1
+        ExecStop=/usr/bin/docker stop nginx1
+        Restart=always
+        RestartSec=5s
+        [Install]
+        WantedBy=multi-user.target
```

Comment:
Is the `ssh_authorized_keys` entry not required?
It would be good to at least have some form of substitution logic shown off in the example file.
This here shows the three types of substitution: template `${name}` for terraform, `$${hostname}` for shell variable substitution (those two are the tricky ones!), and `$(hostname)` for subshell substitution.
```
storage:
  files:
    - path: /home/core/works
      filesystem: root
      mode: 0755
      contents:
        inline: |
          #!/bin/bash
          set -euo pipefail
          hostname="$(hostname)"
          echo My name is ${name} and the hostname is $${hostname}
```

Reply:
You're correct in that it's required - terraform would not be able to log into Flatcar (nor would anyone else) if it was missing.

It was moved into a separate file https://github.com/flatcar/flatcar-terraform/pull/12/files#diff-36a7f9b2bd3991a99d0011e87a0fd95400717e27aa362fceea1eb821b347f328 which is injected into every server config https://github.com/flatcar/flatcar-terraform/pull/12/files#diff-4dc765cd35b30e9e54a37cabe034d0d5b2dc7f3fd3a0591617cc3923b5cddceaR76-R78 to avoid boiler plating.

Reply:
I'm not sure though if the server yamls gain any value from terraform variables.
I wanted to keep the server yamls  as generic as possible, so people can use this without the need to learn terraform.

The terraform literate folks I would expect to take enough from this example to add templating for servers (if they see fit).

Reply:
Ah, I didn't see that the templating is gone now for this file, ok.